python/astro_metadata_translator/observationInfo.py

```diff
@@ -97,6 +113,37 @@ class ObservationInfo:
     Values of the properties are read-only.
     """
 
+    # Static typing requires that we define the standard dynamic properties
+    # statically.
+    if TYPE_CHECKING:
+        telescope: int
+        instrument: str
+        location: astropy.coordinates.EarthLocation
+        exposure_id: int
+        visit_id: int
+        physical_filter: str
+        datetime_begin: astropy.time.Time
+        datetime_end: astropy.time.Time
+        exposure_time: astropy.units.Quantity
+        dark_time: astropy.units.Quantity
+        boresight_airmass: float
+        boresight_rotation_angle: astropy.units.Quantity
+        boresight_rotation_coord: str
+        detector_num: int
+        detector_name: str
+        detector_serial: str
+        detector_group: str
+        detector_exposure_id: int
+        object: str
+        temperature: astropy.units.Quantity
+        pressure: astropy.units.Quantity
+        relative_humidity: float
+        tracking_radec: astropy.coordinates.SkyCoord
+        altaz_begin: astropy.coordinates.AltAz
+        science_program: str
+        observation_type: str
+        observation_id: str
```

Comment:
I wonder if you can do this by adding the types to `ObservationInfo.__annotations__` when the properties are created?

Reply:
I'm fairly certain that mypy does not support `__annotations__` because it does not actually run the python code.

Reply:
@TallJimbo do you have any thoughts on this question as to whether it's possibly for mypy to derive properties?

Reply:
Maybe there's some way to do it via a MyPy plugin, but I'm pretty sure there's nothing you can do short of that that isn't equivalent to what you've done here.